examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjectsStorage.agent.ts

```diff
@@ -10,73 +12,41 @@ import { BpiSubject } from '../models/bpiSubject';
 export class BpiSubjectStorageAgent extends PrismaService {
   async getBpiSubjectById(id: string): Promise<BpiSubject> {
     const bpiSubjectModel = await this.bpiSubject.findUnique({
-      where: { id: id },
+      where: { id },
     });
 
     if (!bpiSubjectModel) {
       throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
     }
-
-    return new BpiSubject( // TODO: Write generic mapper prismaModel -> domainObject
-      bpiSubjectModel.id,
-      bpiSubjectModel.name,
-      bpiSubjectModel.description,
-      bpiSubjectModel.type,
-      bpiSubjectModel.publicKey,
-    );
+    return Mapper.map(bpiSubjectModel, getType<BpiSubject>()) as BpiSubject;
   }
 
   async getAllBpiSubjects(): Promise<BpiSubject[]> {
     const bpiSubjectModels = await this.bpiSubject.findMany();
-    return bpiSubjectModels.map((bp) => {
-      return new BpiSubject(
-        bp.id,
-        bp.name,
-        bp.description,
-        bp.type,
-        bp.publicKey,
-      );
+    return bpiSubjectModels.map((bpiSubjectModel) => {
+      return Mapper.map(bpiSubjectModel, getType<BpiSubject>()) as BpiSubject;
     });
   }
 
   async createNewBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
+    const { name, description, publicKey, type } = bpiSubject;
```

Comment:
why introducing this additional line when the bpiSubject is already containing the params. This is especially cumbersome when there are many params.

Reply:
I'm going to be replacing this whole piece with one liner. Some more code coming in next commit